src/consensus/consensus.rs

```diff
@@ -680,6 +805,15 @@ impl Consensus {
                 // let total_peers = state.consensus.driver.validator_set().count() - 1;
 
                 // println!("Connected to {connected_peers}/{total_peers} peers");
+                match &mut state.shard_validator.block_proposer {
+                    None => {}
+                    Some(block_proposer) => {
+                        match block_proposer.register_validator(&validator).await {
```

Comment:
Let's have the shard validator forward the call to the block proposer. This is looking too deep into the validator.